runtime/domain.c

```diff
@@ -932,32 +924,57 @@ void caml_request_minor_gc (void)
 static void caml_poll_gc_work()
 {
   CAMLalloc_point_here;
+  { // No GC in this block
+    int domain_minor_heap_full = ((uintnat)Caml_state->young_ptr - Bhsize_wosize(Max_young_wosize) <
+        (uintnat)Caml_state->young_start);
+
+    int need_minor_gc = Caml_state->requested_minor_gc;
+
+    // Check if our minor heap is full. If it is then we need to try to grab a
+    // new one from the global minor heap.
+    if( domain_minor_heap_full && !need_minor_gc ) {
+      uintnat global_ptr =
```

Comment:
Is there a reason why we need to do this check with the `global_ptr`? We have to deal with that case inside `caml_replenish_minor_heap` already. 

I didn't understand why the body of this if block wasn't just:
```C
/* try to get a new minor heap */
if(!caml_replenish_minor_heap()) {
   need_minor_gc = 1;
}
```

Reply:
Good catch, this case can be simplified indeed.